pkg/reconciliation/reconcile_racks_test.go

```diff
@@ -1474,3 +1475,219 @@ func Test_callPodEndpoint_RequestFail(t *testing.T) {
 		assert.Fail(t, "Should have returned error")
 	}
 }
+
+func TestCleanupAfterScalingAsyncFeatures(t *testing.T) {
+	rc, cleanupMockScr := setupTestEnv()
+	defer cleanupMockScr()
+	assert := assert.New(t)
+
+	mockHttpClient := &mocks.HttpClient{}
+
+	// Mock features request to support AsyncSSTableTasks
+	mockFeaturesEnabled(mockHttpClient)
+
+	rc.NodeMgmtClient = httphelper.NodeMgmtClient{
+		Client:   mockHttpClient,
+		Log:      rc.ReqLogger,
+		Protocol: "http",
+	}
+
+	mockPodAnnotationsUpdate(rc)
+
+	// Mock async callKeyspaceCleanup
+	mockCallKeyspaceCleanup(mockHttpClient)
+
+	r := rc.cleanupAfterScaling()
+	if r != result.RequeueSoon(2) {
+		t.Fatalf("expected result of result.RequeueSoon(2) but got %s", r)
+	}
+
+	// Mock jobDetails to return success
+	mockJobDetailsResponse(mockHttpClient)
+
+	// Job should be done on the next pass
+	assert.Eventually(func() bool {
```

Comment:
There is no need for `Eventually` here. I tested without multiple times and the test ran fine. I changed line 1519 to be `assert.Equal(r, result.Continue())`.

Reply:
I hadn't looked at the other tests when I added that comment. I went over the other tests and `Eventually` is needed for them. Even if it isn't needed here, it makes sense from a consistency standpoint.